src/components/billing_table/data.ts

```diff
@@ -8,6 +9,30 @@ export const columnConfig: ColType[] = [
   { label: "Total", id: "total", editable: false }
 ]
 
+interface FetchProps {
+  setCollectedRecords: React.Dispatch<React.SetStateAction<ColState[]>>;
+};
+
+export const useFetchData = ({ setCollectedRecords }: FetchProps) => {
+  const [isTableFetching, setIsTableFetching] = React.useState<boolean>(true);
+
+  useEffect(() => {
+      const fetch = () => {
```

Comment:
avoid unnecessary function wrapping, it reduces code complexity. Define the async function directly inside the 'useEffect'